benchmarks/pmap_benchmark.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+
+import numpy as onp
+from tabulate import tabulate
+
+import jax
+import jax.numpy as np
+from jax import pmap
+from jax.util import safe_zip
+
+def pstd(x):
+  return x.std() / x.mean() * 100
+
+def benchmark(f, iters=None, warmup=None, name=None, target_total_secs=None):
+  if target_total_secs is None:
+    target_total_secs = int(os.getenv("TARGET_TOTAL_SECS", 10))
+
+  if warmup is None:
+    if iters is None:
+      warmup = 1
+    else:
+      warmup = onp.clip(1, iters // 10, 10)
+  for _ in range(warmup):
+    f()
+
+  times = []
+  count = 0
+  while (count < iters if iters is not None
+         else sum(times) < target_total_secs):
+    start = time.time()
+    f()
+    end = time.time()
+    times.append(end - start)
+    count += 1
+
+  times = onp.array(times)
+  print("---------Benchmark results for %s---------" % (name or f.__name__))
+  print("mean=%f std=%f %%std=%f total=%f" %
+        (times.mean(), times.std(), pstd(times), times.sum()))
+  print("#iters=%d #warmup=%d" % (count, warmup))
+  print()
+  return times
+
+
+def benchmark_suite(funcs, params_list, param_names, name,
+                    target_total_secs=None):
+  times = []
+  for f, params in safe_zip(funcs, params_list):
+    subname = name + "".join("_%s=%s" % (n, p)
+                             for n, p in safe_zip(param_names, params))
+    times.append(benchmark(f, name=subname,
```

Comment:
I had expected here that f would be invoked with the params. I think this would be clearer if benchmark would be passed both a function and the parameters. To enable the staging of the input data, the benchmark function would be more like the get_benchmark_fn below (signature params -> unit -> unit). WDYT?